services/minespace-web/src/components/common/FileUpload.js

```diff
@@ -59,38 +64,78 @@ const defaultProps = {
   beforeAddFile: () => {},
   beforeDropFile: () => {},
   itemInsertLocation: "before",
+  notificationDisabledStatusCodes: [],
+  shouldReplaceFile: false,
+  replaceFileUploadUrl: "",
+  file: null,
+  shouldAbortUpload: false,
 };
 
 class FileUpload extends React.Component {
   constructor(props) {
     super(props);
 
+    this.state = {
+      file: null,
+    };
+
     this.server = {
       process: (fieldName, file, metadata, load, error, progress, abort) => {
-        const upload = new tus.Upload(file, {
+        if (file) {
+          this.setState({ file: file, progress: progress, load: load });
+        }
+        let fileToUpload = this.state.file ? this.state.file : file;
+        let progressFn = this.state.progress ? this.state.progress : progress;
+        let loadFn = this.state.load ? this.state.load : load;
+
+        const upload = new tus.Upload(fileToUpload, {
           endpoint: ENVIRONMENT.apiUrl + this.props.uploadUrl,
           retryDelays: [100, 1000, 3000],
           removeFingerprintOnSuccess: true,
           chunkSize: this.props.chunkSize,
           metadata: {
-            filename: file.name,
-            filetype: file.type || APPLICATION_OCTET_STREAM,
+            filename: fileToUpload.name,
+            filetype: fileToUpload.type || APPLICATION_OCTET_STREAM,
           },
           headers: createRequestHeader().headers,
           onError: (err) => {
-            notification.error({
-              message: `Failed to upload ${file.name}: ${err}`,
-              duration: 10,
-            });
-            error(err);
+            try {
+              err.response = JSON.parse(err.originalRequest.response);
+              err.originalRequest = err.originalRequest;
+
+              if (
+                !(
+                  this.props.notificationDisabledStatusCodes.length &&
+                  this.props.notificationDisabledStatusCodes.includes(err.response.status_code)
+                )
+              ) {
+                notification.error({
+                  message: `Failed to upload ${
+                    file && fileToUpload.name ? fileToUpload.name : ""
+                  }: ${err}`,
+                  duration: 10,
+                });
+              } else {
+                console.log(
+                  "notification disabled for status code(s):",
+                  this.props.notificationDisabledStatusCodes
+                );
```

Comment:
Please check if this console.log is still required here.

Reply:
Removed this.